core/src/main/scala/kafka/controller/ControllerEventManager.scala

```diff
@@ -83,7 +84,7 @@ class ControllerEventManager(controllerId: Int, rateAndTimeMetrics: Map[Controll
 
     override def doWork(): Unit = {
       queue.take() match {
-        case KafkaController.ShutdownEventThread => initiateShutdown()
+        case KafkaController.ShutdownEventThread =>
```

Comment:
Could we add a comment like "// The shutting down of the thread has been initiated at this point. Just ignore this event"?

Reply:
Yes, sounds good!